Dockerfile

```diff
@@ -6,12 +6,12 @@ LABEL maintainer="BOSCoin Developers <devteam@boscoin.io>"
 COPY ./ /go/src/boscoin.io/sebak
```

Comment:
Using `go install` will be good idea for fast build without `dep ensure` if our project is still in private repo. After this project will be opened, docker image should be applied the latest updated source code from 'master' branch. I think, instead of `COPY` the local source code, `go get boscoin.io/sebak/...` would be natural.

Reply:
I missed something. `go get ...` does not fit our dependencies. I think `dep ensure` should be done.

```
$ time go get boscoin.io/sebak/...
# boscoin.io/sebak/lib/common
src/boscoin.io/sebak/lib/common/util.go:23:29: multiple-value uuid.NewV1() in single-value context
src/boscoin.io/sebak/lib/common/util.go:27:29: multiple-value uuid.NewV4() in single-value context

real	3m45.278s
user	0m35.393s
sys	0m17.996s
```

This is caused by 'github.com/google/uuid', we use '0.2.0' version of uuid, but the master of uuid was updated.

Reply:
It depends what the intended usage is.

This Dockerfile is designed for developer / testers, that's why it lives in the same repository. It provide a very quick edit-compile-test cycle. It allows one to spawn a network quickly because that's part of the cycle: while at the moment we're working on the CLI usage, later on when we work on the node aspect, we can use the same Dockerfile to change one or more instances in a network quickly and feed them messages, to see if our changes make sense.

What you are describing, if I am not mistaken, is a Dockerfile which "just works", provides a clean build, and uses the latest version of the code, I suggest we put that image in another repository, because there's no point in having it within the sebak source tree, since it will be totally independent from the state of the repository.

Does it make sense ?

Reply:
@Geod24 Most of other project to provide `Dockerfile`, the `Dockerfile` in the root means that it can make official image for the project. If you intended this `Dockerfile` to be only for testing, the current location of `Dockerfile` makes to be confused.

Reply:
IMO, Dockerfile should depend on the local source, not GitHub repository. It is common use cases.
And Only pushed docker image is based on latest source code. But we can use automated build on Docker Hub with integrating GitHub.

Reply:
`COPY` cannot access path outside of the build context. If `Dockerfile` did not live at the root of the repository, it wouldn't be able to `COPY` the context (repository). So while an official image can live in a subdirectory, the one used for testing the code in the repo can't.

Reply:
@outsideris Hmm, depending on the local source code, does not guarantee it is based on the latest code or not, so not good way~, I think.

Reply:
@spikeekips : That's exactly what @outsideris mentioned. We're not going to push images based on the root Dockerfile. The one we're going to push will depend on a `Dockerfile` which does a clean build (with `go get` and `dep ensure`).

Reply:
> @Geod24 : COPY cannot access path outside of the build context. If Dockerfile did not live at the root of the repository, it wouldn't be able to COPY the context (repository). So while an official image can live in a subdirectory, the one used for testing the code in the repo can't.

```
$ docker build -f docker/testing/Dockerfile -t SEBAK-testing .
```
This way would be possible with sub directory of `Dockerfile`.

Reply:
> @Geod24 : We're not going to push images to the Docker hub which are based on the root Dockerfile.

Yeah, you already said, this `Dockerfile` is for testing, not publicly opened. I just pointed the location of `Dockerfile`.

Reply:
If `Dockerfile` use source code from GitHub, Docker images can be different by the build timing.
For instance, I build a docker image and build again 5 mins later, two images can be different if we push some commits during the builds. It can be very confused.

Reply:
That's correct, and I did not consider this solution.
I'm not so found of it because it forces you to get more context (the `-f` option is not optional anymore, and Docker's default expects the `Dockerfile` to be at the root of the context) but if we decide to store multiple Docker file in this repository it's a viable solution 👍

Reply:
@outsideris It depends on how we manage the branch to build ;) New version release, new docker image could be released with tags :)